lib/time.ml

```diff
@@ -51,13 +51,13 @@ type sleep = {
 }
 
 module SleepQueue =
-  Lwt_pqueue.Make (struct
+  Binary_heap.Make (struct
                      type t = sleep
-                     let compare { time = t1; _ } { time = t2; _ } = compare t1 t2
+                     let compare { time = t1; _ } { time = t2; _ } = compare t2 t1
                    end)
 
 (* Threads waiting for a timeout to expire: *)
-let sleep_queue = ref SleepQueue.empty
+let sleep_queue = SleepQueue.create 5
```

Comment:
Just so I understand this, instead of a reference you're now creating something else.. (what?) and instead of an empty queue you're preallocating one with 5 slots in it?

Reply:
- The reason for this `ref` is you needed to modify it by adding, removing, ... And `lwt_pqueues` were not mutable, hence the `ref`.
The data structure provided by `bheap` is mutable and modified in-place by these operations ; so there should be no need for a `ref` anymore.
- `empty` is not provided by the `bheap` API as it was with `lwt_pqueue`. Instead, you must use `create` and provide an initial capacity, increased automatically whenever necessary. The value of `5` is arbitrary, you might know better than me if another value is more suitable.

Reply:
To avoid magic constants in `Hashtbl.create` and similar functions, I tend to use `0` - that's less surprising to the eye and it's rarely performance sensitive.

Reply:
I'm afraid this is not possible with `bheap`, the size has to be strictly positive : https://github.com/backtracking/bheap/blob/master/binary_heap.ml#L37-L39

Reply:
`1` ? :)

Reply:
Ok, so `bheap` is exposing an implementation detail to the user. As @emillon said, the actual value doesn't matter much, so feel free to keep `5` (or even `1`) but please add a comment to the effect that it's a workaround for `bheap` requiring a size of `>= 1`.

Reply:
:heavy_check_mark: Done